partner_communication_switzerland/models/partner_communication.py

```diff
@@ -433,7 +434,8 @@ def send_by_sms(self):
             sms_wizard = self.env['sms.sender.wizard'].with_context(
                 partner_id=job.partner_id.id).create({
                     'subject': job.subject,
-                    'text': sms_text
+                    'text': sms_text,
+                    'sms_provider': job.sms_provider
```

Comment:
should be `job.sms_provider_id.id` (integer value is passed for the create method)